### DISCUSSION TITLE
1.26 > 1.27 upgrade failed pretty hard, deleted ALL of my checkpoints

### DISCUSSION THREAD

**Discussion by christianv78** [Acly/krita-ai-diffusion#1331] [General]
I clicked to upgrade from within Krita tonight.  I was simultaneously downloading the new SD 3.5 checkpoint to my Comfy/models folder.  The upgrade took a loooong time and it said Krita wasn't responding for about 15 or more minutes.  Ultimately I got an error in red text saying that there was a problem, and it indicated that there was an issue with the SD 3.5 safetensors file with the .part extension at the end (meaning it wasn't not done downloading).  I thought that was weird, but maybe the 1.27 upgrade also downloaded it?  Oh well.  It said to make sure that the contents of the 1.27 upgrade folder didn't contain any important data and to delete it and try again.  So I did.  Why would it contain important data?  It was a partial failed install after all.  DELETE.  Files are too big to recycle.  Whatever, DELETE.  Then... wait a second.  I go to check my checkpoints in my SEPARATE Comfy UI folder, which is NOT a subfolder of the 1.27 upgrade folder (of course).  And what do I find?  ALL of my checkpoints are GONE.  WTH?  Why???  I'm pretty ticked honestly.  I had a few that are hard to find and they're gone now.  Just gone.  Maybe you need to reconsider how your updater works.

**Comment by noyart** [comment]
Good lesson to always have backups on the stuff you like, anyway. You can try to run testdisk to recover the files, you need extra harddrive ofc to recover the files too, with enough space for the whole disk as it will recover everything, I have only done it with usb drives. This ofc is best to do before you start putting new files on that harddrive, that will overwrite the section with the old files had.

**Reply by christianv78** [reply]
Thanks.  I was able to find them online, I think.  Thankfully it didn't overwrite my Loras or Embeddings.  

I'll just leave this embarrassing story of my stupidity here so others can hopefully see it and avoid doing what I did.

**Comment by Acly** [comment]
Sorry to hear you lost those files. It sounds like an overlap of moving the existing models from the old install into the updated installation (which has to happen at some point), and the browser locking down the folder because it's writing the download at the same time.

To be fair, the updater did not delete those files, but warn you about the situation instead. Maybe it would have helped if it was more specific about potentially containing models?

**Reply by christianv78** [reply]
Thanks.  Yeah, that would probably be helpful.  As a layman, I did not understand that the models were going to be moving anywhere, or really anything that happens under the hood during the upgrade.  It might also be a good idea to warn people to not try downloading any additional content while the upgrade is taking place.  That's probably obvious to some.  Ideally the models could live outside the plugin installation folder, and the plugin could let us point to the place where the models are, so that they are not in harm's way.  

Anyways thanks for the reply, and sorry for the late night angry rant.  I love the plugin for what it's worth.  :)

**Reply by Acly** [reply]
> Ideally the models could live outside the plugin installation folder

That's possible, see https://github.com/Acly/krita-ai-diffusion/wiki/Common-Issues#how-to-share-models-from-another-folder-with-krita-ai-diffusion-plugin

**Comment by artemcrum** [comment]
As I can see, during the server upgrade process the FULL copy of ComfyUI folder is created. Am I right? This seems to be not the optimal way to do it. My ComfyUI folder is >100Gb and I don't even have enough free disk space for a full copy of it. Hope the logic will change)

**Reply by Acly** [reply]
No it does a move. Doesn't require extra space (unless you do something funky and have it on a network share or so)

**Reply by artemcrum** [reply]
Hm... Local setup, Windows machine, nothing funky.

I'm reporting what I actually saw during TOO LONG upgrade process. The upgrade folder was created during the process and somewhere in the middle of it I've discovered same list of checkpoints files in original folder AND upgrade folder.

May be... some folder was locked by another app? Then instead of MOVING script tries to COPY it without any notice?

**Reply by Acly** [reply]
Possible? The script just says to [`move`](https://docs.python.org/3/library/shutil.html#shutil.move) the folder. The documentation does mention something about "copying when source/target or on a different filesystem", but for me it always just moves.

If you want to be sure it's best to keep your models separate via [extra_model_paths.yaml]( https://github.com/Acly/krita-ai-diffusion/wiki/Common-Issues#how-to-share-models-from-another-folder-with-krita-ai-diffusion-plugin)

**Comment by ETCore7** [comment]
Also be aware that if your disk runs out of space and you close Krita your custom brushes and colors and so on will be gone. It happened here while Pinokio was downloading something big in the background. I reported it as a bug at Krita since files should never be deleted, but it was dismissed since Krita did not cause a full disk. So make sure to make backups of your Krita profile as well.
